[no-unused-prop-types] test cases

DianaSuvorova opened the pull request:
fixes #933

ljharb commented on tests/lib/rules/no-unused-prop-types.js:
```diff
@@ -2012,6 +2012,79 @@ ruleTester.run('no-unused-prop-types', rule, {
+      code: [
+        'type ObjectType = {',
+        '  usedDirectly: string;',
+        '  usedFromArray: string;',
```
shouldn't this example only pass when skipShapeProps is false, but fail when it's true?

If so, could we add an explicit config option?

ljharb commented on tests/lib/rules/no-unused-prop-types.js:
```diff
@@ -2012,6 +2012,79 @@ ruleTester.run('no-unused-prop-types', rule, {
+        '  greeting: string;',
+        '};',
+
+        'type Props = AProps & BProps;',
```
ok so this should either mean "name and age and greeting", or "name and age, or greeting" - in the former case, "age" is missing, in the latter case, "greeting" and "name" are mutually exclusive. Shouldn't this be an error?

DianaSuvorova commented:
Whoops, I got carried away, closing too many tickets at once.
None of the cases  I added would work with `skipShapeProps: false` except for #933. Issue #1008 looks like a  legitimate error. I think the rest should be kept open if we want to fix flow props validation.  I updated a PR. thanks!

ljharb commented on tests/lib/rules/no-unused-prop-types.js:
```diff
@@ -2012,6 +2012,23 @@ ruleTester.run('no-unused-prop-types', rule, {
+        ');'
+      ].join('\n'),
+      parser: 'babel-eslint',
+      options: [{skipShapeProps: false}]
```
just to clarify; this example should work identically both with and without skipShapeProps?

DianaSuvorova replied:
yeah, it doesn't have any effect. The props are not shape.
